Refit/RequestBuilderImplementation.cs

```diff
@@ -543,7 +543,21 @@ async Task AddMultipartItemAsync(MultipartFormDataContent multiPartContent, stri
                     // if header, add to request headers
                     if (restMethod.HeaderParameterMap.ContainsKey(i))
                     {
-                        headersToAdd[restMethod.HeaderParameterMap[i]] = paramList[i]?.ToString();
+                        var headerInfo = restMethod.HeaderParameterMap[i];
+                        if (headerInfo.IsHeaderCollection)
+                        {
+                            if (paramList[i] is IDictionary<string, string> headerCollection)
```

Comment:
Seems like this could check for `IEnumerable<string, string>` instead of `IDictionary<string, string>`.

Reply:
There's no `IEnumerable<string, string>` but only `IEnumerable<T>`

Reply:
Sorry, I meant `IEnumerable<KeyValuePair<string, string>>`, which is what `IDictionary` implements.

Reply:
Fixed